tilingManager/tilingLayouts/baseTiling.js

```diff
@@ -237,7 +237,8 @@ var BaseTilingLayout = class BaseTilingLayout {
         ];
         return (
             dialogTypes.includes(metaWindow.window_type) ||
-            !metaWindow.resizeable
+            !metaWindow.resizeable ||
+            Meta.is_wayland_compositor() && metaWindow.get_transient_for() != null
```

Comment:
To reduce potential for future devs having ambiguity around this condition (especially if additional conditions are added at some point) it may be worth adding parenthesis to clearly identify the grouping of `Meta.is_wayland_compositor() && metaWindow.get_transient_for() != null`